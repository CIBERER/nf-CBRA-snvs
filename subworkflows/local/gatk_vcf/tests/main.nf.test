nextflow_workflow {

    name "Test Subworkflow GATK_VCF"
    script "../main.nf"
    workflow "GATK_VCF"

    tag "subworkflows"
    tag "subworkflows/gatk_vcf"
    tag "bwa"
    tag "bwa/mem"
    tag "gatk4"
    tag "gatk4/fastqtosam"
    tag "gatk4/mergebamalignment"
    tag "gatk4/fastqtosam"
    tag "gatk4/baserecalibrator"
    tag "gatk4/applybqr"
    tag "gatk4spark"
    tag "gatk4spark/markduplicates"    
    tag "picard"
    tag "picard/sortsam"


    setup {

        run("PICARD_CREATESEQUENCEDICTIONARY") {
            script "../../../../modules/nf-core/picard/createsequencedictionary/main.nf"
            process {
                """
                input[0] = [
                    [ id: 'genome'],
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true)
                ]
                """
            }
        }
    }

    test("gatk_vcf - bam") {


        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id : 'test_sample' ],
                    file("/mnt/tblab/yolanda/GT-Bioinfo/test/work/64/3b6a740cc15effafee1d72a1c21bbe/Sample1_T1.bam", checkIfExists: true),
                    file("/mnt/tblab/yolanda/GT-Bioinfo/test/work/64/3b6a740cc15effafee1d72a1c21bbe/Sample1_T1.bai", checkIfExists: true)
                ])
                input[1] =  Channel.fromList([
                        tuple([ id: 'genome'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true))
                    ])
                input[2] =  Channel.fromList([
                        tuple([ id: 'genome'],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta.fai', checkIfExists: true))
                    ])
                input[3] = PICARD_CREATESEQUENCEDICTIONARY.out.reference_dict
                input[4] =  Channel.fromList([
                        tuple([ id: 'test_sample'],
                        [])
                    ])
                input[5] = input[5] =  Channel.fromList([
                            tuple([ id: 'test_dbsnp'],
                            [])
                        ])
                input[6] = input[5] =  Channel.fromList([
                            tuple([ id: 'test_dbsnp'],
                            [])
                        ])
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

}
