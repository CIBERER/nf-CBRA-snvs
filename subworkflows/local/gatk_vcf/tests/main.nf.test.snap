{
    "gatk_vcf - bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.biallelic.gatk.vcf.gz:md5,67cc4cf602d4886d01be208ad6564613",
                        "test_sample.biallelic.gatk.vcf.gz.tbi:md5,e1a29bf3619de71aef8ea5591ef7c1ae"
                    ]
                ],
                "1": [
                    "versions.yml:md5,107e9a4b9ace28302e69db3283165da9",
                    "versions.yml:md5,13e78564a0c5c77655ccd7ca568d2865",
                    "versions.yml:md5,1b3410e766f7dd8712f5a5661a683df4",
                    "versions.yml:md5,28024a7d5c6c9ee94118e63b87531234",
                    "versions.yml:md5,4068da5aa3ac02ba8b74d2269c8a1b63",
                    "versions.yml:md5,4e615418b9b828a5c2eb585a0b82ef0b",
                    "versions.yml:md5,507f6f7a60642435ba87d067d1397f49",
                    "versions.yml:md5,5ed3b44c225e548f3b20550c9bab1a9b",
                    "versions.yml:md5,71186879c23e4734ed71ccf69551db39",
                    "versions.yml:md5,9b3adaa352d73e199b859c1b1d66525d",
                    "versions.yml:md5,fc4cd74490d37fbdf6bc4571130ae6e0"
                ],
                "vcf": [
                    [
                        {
                            "id": "test_sample"
                        },
                        "test_sample.biallelic.gatk.vcf.gz:md5,67cc4cf602d4886d01be208ad6564613",
                        "test_sample.biallelic.gatk.vcf.gz.tbi:md5,e1a29bf3619de71aef8ea5591ef7c1ae"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,107e9a4b9ace28302e69db3283165da9",
                    "versions.yml:md5,13e78564a0c5c77655ccd7ca568d2865",
                    "versions.yml:md5,1b3410e766f7dd8712f5a5661a683df4",
                    "versions.yml:md5,28024a7d5c6c9ee94118e63b87531234",
                    "versions.yml:md5,4068da5aa3ac02ba8b74d2269c8a1b63",
                    "versions.yml:md5,4e615418b9b828a5c2eb585a0b82ef0b",
                    "versions.yml:md5,507f6f7a60642435ba87d067d1397f49",
                    "versions.yml:md5,5ed3b44c225e548f3b20550c9bab1a9b",
                    "versions.yml:md5,71186879c23e4734ed71ccf69551db39",
                    "versions.yml:md5,9b3adaa352d73e199b859c1b1d66525d",
                    "versions.yml:md5,fc4cd74490d37fbdf6bc4571130ae6e0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-05-05T08:27:41.614751393"
    }
}